packages/building-blocks/src/stories/color-stories/ColorPalette.story.tsx

```diff
@@ -0,0 +1,231 @@
+      : ""};
+  display: flex;
+  height: 40px;
+  justify-content: space-between;
+  margin: 0;
+  padding: 10px;
+  transition: all 0.2s ease-in;
+  width: 300px;
+
+  &:first-of-type {
+    border-radius: 8px 8px 0 0;
+  }
+  &:last-of-type {
+    border-radius: 0 0 8px 8px;
+  }
+
+  &:hover {
+    transform: scale(1.012);
+    transition: all 0.2s ease-in;
+    cursor: pointer;
+  }
+`;
+
+const ColorPickForm = styled.form`
+  display: flex;
+  gap: 15px;
+  margin-bottom: 15px;
+`;
+
+const WCAGContainer = styled.div`
+  max-width: 425px;
+`;
+
+const ColorPalette = ({ color, label }: Props): JSX.Element => {
+  const colorArray: Array<Hue> = Object.entries(color);
+  const [selectedColor, setSelectedColor] = useState(colorArray[0]);
+  const [foregroundColor, setForegroundColor] = useState(colorArray[0][1]);
+  const [foregroundColorInput, setForegroundColorInput] = useState(
+    colorArray[0][1]
+  );
+  const [backgroundColorInput, setBackgroundColorInput] = useState("#FFFFFF");
+  const [backgroundColor, setBackgroundColor] = useState("#FFFFFF");
+  const [foregroundError, setForegroundError] = useState(false);
+  const [backgroundError, setBackgroundError] = useState(false);
+
+  const REX_EXP = /(^#[0-9A-F]{6}$)/i;
+
+  const isValidHexCode = code => {
+    let hexcode = code;
+    if (!code.startsWith("#")) hexcode = `#${code}`;
+    return REX_EXP.test(hexcode);
+  };
+
+  const backgroundHandleChange = event => {
+    const value = event.target.value;
+    setBackgroundColorInput(value);
+  };
+
+  const foregroundHandleChange = event => {
+    const value = event.target.value;
+    setForegroundColorInput(value);
+  };
+  const hexArray = colorArray.map(x => x[1]);
+  const foregroundColorIsInPalette = () => {
+    if (foregroundColorInput && hexArray.includes(foregroundColorInput)) {
+      return `${label} ${
+        colorArray[hexArray.indexOf(foregroundColorInput)][0]
+      }`;
+    }
+    return null;
+  };
+  const backgroundColorIsInPalette = () => {
+    if (backgroundColorInput && hexArray.includes(backgroundColorInput)) {
+      return `${label} ${
+        colorArray[hexArray.indexOf(backgroundColorInput)][0]
+      }`;
+    }
+    return null;
+  };
+  const selectedColorReset =
+    selectedColor &&
+    foregroundColorInput !== selectedColor[1] &&
+    backgroundColorInput !== selectedColor[1];
+
+  useEffect(() => {
+    if (selectedColorReset) {
+      setSelectedColor(undefined);
+    }
+    if (!isValidHexCode(foregroundColorInput)) {
+      setForegroundError(true);
+    } else {
+      setForegroundError(false);
+      setForegroundColor(foregroundColorInput);
+    }
+  }, [foregroundColorInput]);
+
+  useEffect(() => {
+    if (selectedColorReset) {
+      setSelectedColor(undefined);
+    }
+    if (!isValidHexCode(backgroundColorInput)) {
+      setBackgroundError(true);
+    } else {
+      setBackgroundError(false);
+      setBackgroundColor(backgroundColorInput);
+    }
+  }, [backgroundColorInput]);
```

Comment:
Why is this in here twice? Also, how is this using variables that aren't in the variable array?